From: Jan Kara <jack@suse.cz>
Subject: Fix kernel hang during cluster initialization
Reference: 300730
Patch-mainline: Never (mainline does not have this code)

Fix hang during cluster initialization due to uninitialized rwlock.

Signed-off-by: Jan Kara <jack@suse.cz>

--- a/fs/ocfs2/cluster/tcp.c.orig	2007-10-09 16:05:14.000000000 +0200
+++ b/fs/ocfs2/cluster/tcp.c	2007-10-09 15:57:52.000000000 +0200
@@ -129,7 +129,7 @@ static struct work_struct o2net_listen_w
 static struct o2hb_callback_func o2net_hb_up, o2net_hb_down;
 #define O2NET_HB_PRI 0x2
 
-static struct blocking_notifier_head o2net_notifier_head;
+static BLOCKING_NOTIFIER_HEAD(o2net_notifier_head);
 
 static struct o2net_handshake *o2net_hand;
 static struct o2net_msg *o2net_keep_req, *o2net_keep_resp;
